### ISSUE TITLE
Dynamic enablement of access log results in plain text logs

### ISSUE THREAD

**Issue by a-thaler** [kyma-project/kyma#14909]
**Description**

The Kyma production profile disables istio access logs by default, having the default log format configured with a JSON format and encoding. When enabling in such setup the access logs dynamically using the new telemetry API of istio as described in https://kyma-project.io/docs/kyma/latest/04-operation-guides/operations/obsv-03-enable-istio-access-logs/, then all should work as expected.
However, with Kyma 2.4 it looks like that the JSON format is gone and the access logs are in plain text. 

As part of the istio 1.14 release the telemetry feature got refactored (see https://istio.io/latest/news/releases/1.14.x/announcing-1.14/#telemetry-improvements) and potentially the default extension provider got changed.

**Expected result**
Have a cluster with production profile. Apply following resource:
```
apiVersion: telemetry.istio.io/v1alpha1
kind: Telemetry
metadata:
  name: access-config
  namespace: kyma-system
spec:
  accessLogging:
    - providers:
      - name: envoy
 ```
then the istio-proxy logs of the logging-fluent-bit containers should have logs printed like:
```
{"upstream_host":"XXX","downstream_local_address":"XXX:3100","start_time":"2022-07-25T14:47:12.028Z","upstream_service_time":"2","duration":2,"requested_server_name":null,"method":"POST","response_code_details":"via_upstream","response_flags":"-","upstream_cluster":"outbound|3100||logging-loki.kyma-system.svc.cluster.local","path":"/loki/api/v1/push","response_code":204,"upstream_transport_failure_reason":null,"authority":"logging-loki:3100","route_name":"default","protocol":"HTTP/1.1","request_id":"7aea1e76-ad41-484e-992c-2fc1e3739ad2","downstream_remote_address":"XXX:33200","x_forwarded_for":null,"connection_termination_details":null,"upstream_local_address":"XXX:33458","bytes_sent":0,"user_agent":"promtail/","bytes_received":4160}

```

**Actual result**
```
[2022-07-25T15:40:11.028Z] "POST /loki/api/v1/push HTTP/1.1" 204 - via_upstream - "-" 2837 0 2 2 "-" "promtail/" "8981175e-1f89-9f09-82bf-dd32c5970b6e" "logging-loki:3100" "XXX:3100" outbound|3100||logging-loki.kyma-system.svc.cluster.local XXX:38480 XXX:3100 XXX:35424 - default
```

**Steps to reproduce**

**Troubleshooting**

**Comment by a-thaler** [comment]
Indeed, with istio 1.14 the "envoy" extensionProvider is not defined by default anymore. Adding an extensionProviders section to the meshconfig fixes the problem:

```yaml
extensionProviders:
    - name: envoy
      envoyFileAccessLog:
        path: "/dev/stdout"
        logFormat:
          labels: {}
```
Defining it as defaultProvider:
```yaml
  defaultProviders:
    accessLogging:
      - envoy
```
will make the setting in the telemetry resource obsolete:
```yaml
apiVersion: telemetry.istio.io/v1alpha1
kind: Telemetry
metadata:
  name: logs
  namespace: kyma-system
spec:
  accessLogging:
    - disabled: false
  ```

**Comment by a-thaler** [comment]
Fixed for Kyma 2.5

**Comment by a-thaler** [comment]
Test automation will be added by https://github.com/kyma-project/kyma/issues/14925
